### DISCUSSION TITLE
Return Output from Python Scripting Modules in Bonsai

### DISCUSSION THREAD

**Discussion by jaroducsb** [bonsai-rx/bonsai#2005] [Q&A] [answered]
Hello!

I just started using the Python Scripting package and am hoping to use outputs from the scripts I'm importing to use in Bonsai. Would this be possible/feasible?

**Comment by ncguilbeault** [comment, marked as answer]
Hi,

Yes, you can share variables between python and Bonsai. When you use `Get` to access a variable inside of python, you can treat the output like a string and transform the output into other data types. For example, if your python variable is convertible to a double, you can use an `ExpressionTransform` and call `Convert.ToDouble(it)`, which will output a double. I've attached a simple workflow to demonstrate this: [python-conversion.zip](https://github.com/user-attachments/files/16913269/python-conversion.zip). For more complex data types, you may want to use a tool like [Bonsai.Sgen](https://github.com/bonsai-rx/sgen), or you can check out how we transform python data types in the [Bonsai.ML package](https://bonsai-rx.org/machinelearning/).

**Reply by jaroducsb** [reply]
Worked wonderfully, thank you so much! I just peeked at the ML package and I think I will be diving into that for a future project we'll be working on. Exciting stuff, thank you again for the assistance!
